.travis.yml

```diff
@@ -1,10 +1,43 @@
 # See Dockerfile for the full build instructions
 sudo: required
-language: cpp
+language: python
+cache: pip
+
+#mainly for flake8 as everything else runs in docker
+python: 
+  - 3.5
+
+git:
+  quiet: true
 
 services:
   - docker
 
-script:
-  - docker build .
-
+jobs:
+  include:
+    - stage: PythonTests
+      install: pip install flake8
+      script: flake8 --exit-zero cocotb/
+    - stage: SimTests
+      # env only here for easier visibility in travis-ci
+      env:
+        - COCOTB_PY27
+      before_install: &docker_prep
+        - docker build -t cocotb .
+        - docker images -a
+        - docker run -d --name cocotb cocotb tail -f /dev/null
+        - docker ps -a
+      script:
+        - echo 'Running cocotb tests with python 27 ...' && echo 'travis_fold:start:cocotb_py27' 
+        - docker exec -it -e TRAVIS=true -e COCOTB=/build-py2/src cocotb bash -lc 'source /build-py2/venv/bin/activate; make -C /build-py2/src test'
+        - echo -e 'travis_fold:end:cocotb_py27'
+    - stage: SimTests
+      # env only here for easier visibility in travis-ci
+      env:
+        - COCOTB_PY35
+      before_install: *docker_prep 
+      script:
+        - echo 'Running cocotb tests with python 35' ... && echo -en 'travis_fold:start:cocotb_py35'
```

Comment:
nit: ... "with Python 3.5"

You also want to include the "..." into the single quotes (or skip them alltogether), and make the second echo a new command in the travis ci file.

Reply:
the second echo has to be on the same line otherwise the script will just exit with 0. 
it kinda serves as anchor for the folding label.